transformer_engine/pytorch/__init__.py

```diff
@@ -14,13 +14,8 @@
 from transformer_engine.common import load_framework_extension
 
 
-@functools.lru_cache(maxsize=None)
-def torch_version() -> tuple[int, ...]:
-    """Get PyTorch version"""
-    return PkgVersion(str(torch.__version__)).release
-
-
-assert torch_version() >= (2, 1), f"Minimum torch version 2.1 required. Found {torch_version()}."
+torch_version = PkgVersion(str(torch.__version__)).release
+assert torch_version >= (2, 1), f"Minimum torch version 2.1 required. Found {torch_version}."
```

Comment:
Please change the instances at the call site as well. For e.g. `linear.py` still calls `torch_version()`